scripts/build-js/heroicons-js-base.js

```diff
@@ -0,0 +1,43 @@
+const heroiconsLoader = {
+  replaceAllIcons: () => {
+    const getAttrs = (element) =>
+    Array.from(element.attributes).reduce((attrs, attr) => {
+      attrs[attr.name] = attr.value;
+      return attrs;
+    }, {});
+
+    const replaceElement = (element) => {
+      const attributes = getAttrs(element);
```

Comment:
You are converting all attributes to an object to just get two data attributes out of it. This seem to be unnecessary. You could just use `element.getAttribute('data-heroicons')`. 

Or a more modern variant would be to use the elements dataset `element.dataset.heroicons`. Here a singular `heroicon` would make sense.

Reply:
Thanks for this suggestion. You're totally right, a lot of this code was totally redundant. I've refactored as you suggested.